zebra/zebra_vty.c

```diff
@@ -783,9 +795,11 @@ DEFPY(ip_route_address_interface,
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
-	if (nexthop_vrf)
+	if (nexthop_vrf) {
 		nh_zvrf = zebra_vrf_lookup_by_name(nexthop_vrf);
-	else
+		if (!nh_zvrf)
+			nh_zvrf = zebra_vty_get_unknown_vrf(vty, nexthop_vrf);
```

Comment:
rather than repeat this pattern in multiple places, wouldn't it be better to just add a flag to zebra_vrf_lookup_by_name to allocate vrf (i.e., call zebra_vty_get_unknown_vrf ) on lookup failure?

Reply:
I thought about this and I did not want lookup to create in all cases.

Reply:
fixed